### PR TITLE
Configure outbound load balancers for azure private clusters

### PR DESCRIPTION
**Pull request by saimanoj01**
### What this PR does / why we need it
Add ability to configure private clusters on Azure. 
- Keep the definition of private cluster the same with clusters created in TKG 1.4 and TKG 1.5 which is to create the outbound load balancers for control plane and worker VMs.
- Add new variables to disable creating outbound load balancers.

```
AZURE_ENABLE_CONTROL_PLANE_OUTBOUND_LB: true
AZURE_ENABLE_NODE_OUTBOUND_LB: true
AZURE_CONTROL_PLANE_OUTBOUND_LB_FRONTEND_IP_COUNT: 1
AZURE_NODE_OUTBOUND_LB_FRONTEND_IP_COUNT: 1
#! specifies the number of minutes to keep a TCP connection open for the outbound rule
#! https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-tcp-reset#configurable-tcp-idle-timeout
AZURE_NODE_OUTBOUND_LB_IDLE_TIMEOUT_IN_MINUTES: 4
```

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
1. Created an Azure private cluster and made sure that it is created with outbound load balancers.
2. Created an Azure private cluster by disabling the outbound load balancers. This is verified by setting up a squid proxy on a VM with internet access created on the same VNET as the private cluster.

### Release note

```release-note
Add config variables to disable outbound load balancers for Azure private clusters.
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/infrastructure-azure/v1.0.0/ytt/overlay.yaml [hunk 1]
```diff
@@ -119,6 +119,20 @@ spec:
...
+    #@overlay/match missing_ok=True
+    controlPlaneOutboundLB:
+      frontendIPsCount: #@ data.values.AZURE_CONTROL_PLANE_OUTBOUND_LB_FRONTEND_IP_COUNT
+    #@ end
+
+    #@ if data.values.AZURE_ENABLE_NODE_OUTBOUND_LB:
+    #@overlay/match missing_ok=True
+    nodeOutboundLB:
+      frontendIPsCount: #@ data.values.AZURE_NODE_OUTBOUND_LB_FRONTEND_IP_COUNT
+      idleTimeoutInMinutes: #@ data.values.AZURE_NODE_OUTBOUND_LB_IDLE_TIMEOUT_IN_MINUTES
```
Comment (shysank) [inline pkg/v1/providers/infrastructure-azure/v1.0.0/ytt/overlay.yaml:133]: is this needed?

**Review by shysank — COMMENTED** [review]
/lgtm with one small clarification

File: pkg/v1/providers/infrastructure-azure/v1.0.0/ytt/overlay.yaml (hunk 1 above)
Comment (saimanoj01) [reply pkg/v1/providers/infrastructure-azure/v1.0.0/ytt/overlay.yaml:133]: Upstream documentation - https://capz.sigs.k8s.io/topics/node-outbound-lb.html

```
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: AzureCluster
metadata:
  name: my-public-cluster
  namespace: default
spec:
  location: eastus
  networkSpec:
    apiServerLB:
      type: Public
    nodeOutboundLB:
      frontendIPsCount: 3
      idleTimeoutInMinutes: 4
```

**Review by imikushin — APPROVED** [review]
LGTM

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**